axlearn/common/checkpointer.py

```diff
@@ -922,3 +953,220 @@ def validate_and_restore(*, step: int, ckpt_dir: str):
             restored_state = state
 
         return step, restored_state
+
+
+class OrbaxCheckpointer(BaseCheckpointer):
+    """A checkpointer that uses orbax CheckpointManager.
+
+    NOTE: While this class uses index files to do additional validation on checkpoint state, the
+    index file is not used for committing checkpoints (i.e., it is not reliable to check for the
+    presence of 'index' file). Instead, use `checkpoint_paths` to identify committed checkpoints.
+    """
+
+    # Disable lazy imports within the class, to avoid requiring orbax dependency globally.
+    # pylint: disable=import-outside-toplevel
+
+    @config_class
+    class Config(BaseCheckpointer.Config):
+        """Configures OrbaxCheckpointer.
+
+        Attributes:
+            keep_last_n: Keep this many past ckpts.
+            validation_type: Checkpoint validation during restore.
+            save_concurrent_gb: Max concurrent GB during save. Defaults to 96 (orbax default).
+            restore_concurrent_gb: Max concurrent GB during write. Defaults to 96 (orbax default).
+        """
+
+        keep_last_n: int = 1
+        validation_type: CheckpointValidationType = CheckpointValidationType.EXACT
+        save_concurrent_gb: int = 96
+        restore_concurrent_gb: int = 96
+
+    @classmethod
+    def checkpoint_paths(cls, base_dir: str) -> List[str]:
```

Comment:
Is this method ever called?

Reply:
Not in this module (but by other code as part of its API).